### ISSUE TITLE
Dependendecy on net.kyori.adventure-platform-bukkit/4.0.0-SNAPSHOT does not exist

### ISSUE THREAD

**Issue by mdsimmo** [PlaceholderAPI/PlaceholderAPI#741]
### Confirmation

- [x] My issue isn't already found on the Issue tracker.
- [X] My issue is about **PlaceholderAPI** and not any expansion or external plugin
- [X] The issue isn't already fixed in a Spigot Release or Development Build.

### Type

Plugin Bug

### What happens?

Cloning and building this repository fails builds since https://repo.maven.apache.org/maven2/net/kyori/adventure-platform-bukkit/ does not have `4.0.0-SNAPSHOT`. This used to work, so I'm guessing that the snapshot has been deleted and 4.0.0 is now used (kinda strange that they would delete it though)?

Propose fix of updating adventure platform to `4.0.0` instead.

### Expected Behaviour

Should build without errors

### How to Reproduce

``` bash
git clone https://github.com/PlaceholderAPI/PlaceholderAPI.git
cd PlaceholderAPI
./gradlew build
```

### `/papi dump` Output

-

### Errors

https://paste.helpch.at/fuhifajemi.sql

**Comment by Andre601** [comment]
I doubt this would change anything, but did you try executing `./gradlew clean shadowJar` instead of `./gradlew build`?

**Comment by Andre601** [comment]
Some progress.
It seems the issue was/is, that the URL for the CodeMC repository was missing a `/` at the end and Gradle was a bit picky about this.
I still updated the dependencies in my PR. Could you try out the [`feature/update-dependencies` branch](https://github.com/PlaceholderAPI/PlaceholderAPI/tree/feature/update-dependencies) and let me know if compiling (and running) the plugin works for you? Thanks.

**Comment by spnda** [comment]
Just to note, Kyori seem to have just removed 4.0.0-SNAPSHOT from the [sonatype snapshot repository](https://oss.sonatype.org/content/repositories/snapshots/net/kyori/adventure-platform-bukkit/), breaking the build process unless you have a local build of it in your m2 repository.
Also, yes, obviously, your PR will fix this issue as it is referencing an existing version of the artifact. Additionally, it might be worth updating Gradle to 7.3 while you're at it updating stuff.

**Comment by mdsimmo** [comment]
@Andre601 
I tried the updated-dependencies branch, and this is the result:
* It complains about invalid license files, which I can fix by removing the license checks.
* If using JDK 17, it complains about  "Unsupported class file major version 61", which I can fix by downgrading to jdk 8:
* Otherwise, yes, it fixes the original issue as @spnda points out

**Comment by spnda** [comment]
For your issue with JDK 17:
PlaceholderAPI uses Gradle 7.0 iirc, which doesn't support JDK 17. You can either fix this by using JDK 16 or changing the version in `gradle/wrapper/gradle_wrapper.properties` to Gradle 7.3.

**Comment by darbyjack** [comment]
Closed via https://github.com/PlaceholderAPI/PlaceholderAPI/commit/acb5728e65737ad819ab4380fa5049d16a350738

**Comment by spnda** [comment]
I just wanted to note that this issue has not really been fixed for people who depend on PlaceholderAPI. The [maven](https://repo.extendedclip.com/content/repositories/placeholderapi/me/clip/placeholderapi/2.10.10/) has still not gotten an updated version and I don't think anyone should directly depend on the [dev builds](https://repo.extendedclip.com/content/repositories/dev/me/clip/placeholderapi/), which are in fact updated. Will we get a 2.10.11 anytime soon with the fixed builds?

**Comment by darbyjack** [comment]
We are shooting to release the next version by next week.

**Comment by spnda** [comment]
I hope it's as soon as possible as my CIs have been broken for over a month now just because of this 😄

**Comment by mdsimmo** [comment]
It's not a great solution, but I "fixed" my CI by using version 2.9 which didn't depend on adventure-platform

**Comment by Andre601** [comment]
I believe one way you could do this for now is using [jitpack](https://jitpack.io) to get the latest state of the `development` branch, which should have the updated dependencies of PAPI in it.

It's not an ideal solution, but as a temporary workaround should it work.

Direct link for download info:
https://jitpack.io/#PlaceholderAPI/PlaceholderAPI/development-SNAPSHOT
